src/messaging.rs

```diff
@@ -243,4 +264,54 @@ impl<NodeUid: Clone + Ord> NetworkInfo<NodeUid> {
     pub fn is_validator(&self) -> bool {
         self.is_validator
     }
+
+    /// Returns `true` if the given node takes part in the consensus itself. If not, it is only an
+    /// observer.
+    pub fn is_node_validator(&self, uid: &NodeUid) -> bool {
+        self.public_keys.contains_key(uid)
+    }
+
+    /// Generates a map of matching `NetworkInfo`s for testing.
```

Comment:
Maybe add `#[cfg(test)]`?

Reply:
Everything else looks good though!

Reply:
Then the build fails for me. I think `#[cfg(test)]` is only included for the `--lib`'s own unit tests, but not available to the integration tests or the `simulation` example.